### DISCUSSION TITLE
Returning "insufficient_scope" after changing legacy tocken

### DISCUSSION THREAD

**Discussion by Devarsh-Sinalite** [etsy/open-api#902] [Q&A]
Hi,

I call legacy token exchange api and it returns access token but when i use the same token for getShopShippingProfile, it returns insufficient_scope error. earlier with apiv2 it was working ok and it had all scope and was able to retrieve shipping profiles.

can anyone help how can i exchange token with all scope?

Thanks,

Devarsh

**Comment by Devarsh-Sinalite** [comment]
can anybody help me with this?

**Comment by PMStanley** [comment]
You can confirm what scopes your tokens have using the tokenScopes endpoint: https://developer.etsy.com/documentation/reference/#operation/tokenScopes

If the scopes on your new OAuth2 token don't match those of the old OAuth1 you should reach out to Etsy directly via email because the documentation states the new tokens will have the same scope as the old ones. In the mean time you can get the affected user(s) to go through the auth flow again with the correct set of scopes.

**Reply by Devarsh-Sinalite** [reply]
@PMStanley Thank you so much for your reply. I checked and found that OAuth1 get shipping profiles(**findAllUserShippingProfiles**) are under **listings_r** scope but in OAuth2 it is under **shops_r**.  Now i am not sure how can i solve this situation without auth flow for affected users. user's woudn't like to go through the auth flow again and again. 

@etsyachristensen  any other solution for this?
![image](https://user-images.githubusercontent.com/74913466/224517569-fda3fc09-5b3c-4f70-9aa5-59b6cf5cbfdb.png)
![image](https://user-images.githubusercontent.com/74913466/224517594-cef4eef0-3c7c-4e0a-89e0-5a842766ca8d.png)

**Reply by PMStanley** [reply]
I think there's only two options here:

1) You get the users to go through the auth flow again with the set of scopes you require (good opportunity to include any you need for future developments here too). If you word it right you should hopefully see minimal drop out, though I would anticipate that some users will not complete the process.
2) Etsy assign the shops_r scope to your new tokens at the backend.  The issue I see here is it's work for Etsy and their argument may be that customers should reconfirm their desire for your app to have access to their shop.

You may get lucky with option 2 but you'll need to reach out via email and a response may take some time.  Personally I would start to look at option 1 in the mean time and try to implement it in the least disruptive way to your users.

**Reply by Devarsh-Sinalite** [reply]
@PMStanley sure, make sense to start implementing #1 as i know it will take some time to get response for #2 from Etsy. 

I really appreciate your help. Thank you!
